### PR TITLE
Support for the new `watchos` platform

### PR DESCRIPTION
**Pull request by neonichu**
Fixes some issues with the newly added `watchos` platform. This lays the groundwork for supporting watchOS as a new platform in CocoaPods.

Things yet to be addressed:

- [x] Specs
- [x] Changelog
- [x] Pods integrated into `watchapp2` and `watchkit2-extension` targets should automatically be build for watchOS
- [x] Add watchOS example (unused for now, because Travis doesn't have Xcode 7)

See also CocoaPods/Core#249 and CocoaPods/Xcodeproj#272

### CODE REVIEW

**Comment by neonichu** [comment]
See <https://github.com/neonichu/native-watchOS-example> for an example on how to try this out.

File: lib/cocoapods/generator/header.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ def save_as(path)
       # @return [String]
       #
       def generate_platform_import_header
-        "#import #{platform == :ios ? '<UIKit/UIKit.h>' : '<Cocoa/Cocoa.h>'}\n"
+        "#import <Foundation/Foundation.h>\n"
```
Comment (segiddins) [inline lib/cocoapods/generator/header.rb]: This is a backwards incompatible change?

Comment (neonichu) [inline lib/cocoapods/generator/header.rb]: Possibly - might be a better idea to only use `Foundation` for `watchos` targets.

**Comment by mrackwitz** [comment]
Regarding specs:
Once Travis supports Xcode 7, we should definitively add an example for watchOS to ensure that our produced integrations can be actually build on that platform as well. But you could add an example already now, just ensure that it doesn't fail `rake examples:build` when Xcode 7 is not available, so we can test that at least locally.
Beside that we should cover that also by an integration spec, I think we could update e.g. `install_multiple_targets`.

**Comment by neonichu** [comment]
@mrackwitz Will add an example, not sure what the integration spec should actually test, though. Since the changes on this repo are minimal, it seems a bit redundant.

File: lib/cocoapods/generator/header.rb [hunk 2]
```diff
@@ -82,7 +82,11 @@ def save_as(path)
       # @return [String]
       #
       def generate_platform_import_header
-        "#import #{platform == :ios ? '<UIKit/UIKit.h>' : '<Cocoa/Cocoa.h>'}\n"
+        case platform.name
+        when :ios then "#import <UIKit/UIKit.h>\n"
+        when :osx then "#import <Cocoa/Cocoa.h>\n"
+        else "#import <Foundation/Foundation.h>\n"
```
Comment (kylef) [inline lib/cocoapods/generator/header.rb]: Perhaps we should leave this empty for new platforms?

Re: #1746

Comment (segiddins) [inline lib/cocoapods/generator/header.rb]: :+1:

Comment (neonichu) [inline lib/cocoapods/generator/header.rb]: Good idea :+1:

Comment (mrackwitz) [inline lib/cocoapods/generator/header.rb]: Foundation must be imported as long as the `FOUNDATION_EXPORT` macro is used by the umbrella header.

**Comment by neonichu** [comment]
> Pods integrated into watchapp2 and watchkit2-extension targets should automatically be build for watchOS

Seems like this is generally not done and it is up to the user to define the correct platform for the watchOS targets.

**Comment by segiddins** [comment]
@neonichu isn't that done via target introspection already?

**Comment by neonichu** [comment]
@segiddins I think it's supposed to be done, but doesn't actually function.

**Comment by neonichu** [comment]
OK, this should be merge-able now :tada:

**Comment by segiddins** [comment]
Should already be done in `TargetInspector#compute_platform`

**Comment by segiddins** [comment]
:+1:

**Comment by neonichu** [comment]
:watch: :tada:

File: lib/cocoapods/generator/header.rb (hunk 2 above)
Comment (kylef) [inline lib/cocoapods/generator/header.rb]: In that case, shouldn't the umbrella header import Foundation and not the pch file?

**Comment by songjianboy** [comment]
I would like to move watch target's resource and storyboard as a pod library, but when I
compile error “error: watchOS storyboards do not support target device type "mac””。
I found Pods-JDWatch-resources.sh script TARGET_DEVICE_ARGS not include watch，Is cocoaPods not supported like this？please help me！

**Pull request merged**